tiledb/sm/array_schema/enumeration.cc

```diff
@@ -235,14 +243,24 @@ shared_ptr<const Enumeration> Enumeration::extend(
     uint64_t data_size,
     const void* offsets,
     uint64_t offsets_size) const {
-  if (data == nullptr) {
-    throw EnumerationException(
-        "Unable to extend an enumeration without a data buffer.");
-  }
+  if (data == nullptr && data_size == 0) {
+    // This is OK
+    //
+    // Users need to be able to create an enumeration containing
+    // one value which is the empty string.
+    //
+    // The std::vector<uint8_t> foo(0, 0) constructor at our callsite results
+    // in a foo.data() which is nullptr and foo.size() which is zero.
+  } else {
+    if (data == nullptr) {
+      throw EnumerationException(
+          "Unable to extend an enumeration without a data buffer.");
+    }
 
-  if (data_size == 0) {
-    throw EnumerationException(
-        "Unable to extend an enumeration with a zero sized data buffer.");
+    if (data_size == 0) {
+      throw EnumerationException(
+          "Unable to extend an enumeration with a zero sized data buffer.");
+    }
```

Comment:
We can simplify it to `if (data_size == 0) {} if (data == nullptr)`, and also add an `&& !var_size()` to the first condition.

Reply:
@teo-tsirpanis can you please suggest a patch? I'm not quite following what you want me to do here. Thank you!

Reply:
Sorry I got confused and can't think of how the code would look like. I will approve and we can revisit this later.